container/src/main/java/org/openremote/container/web/DynamicTimeInjectionFilter.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2017, OpenRemote Inc.
+ *
+ * See the CONTRIBUTORS.txt file in the distribution for a
+ * full listing of individual contributors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.openremote.container.web;
+
+import jakarta.annotation.Priority;
+import jakarta.ws.rs.client.ClientRequestContext;
+import jakarta.ws.rs.client.ClientRequestFilter;
+import jakarta.ws.rs.core.MultivaluedHashMap;
+import jakarta.ws.rs.core.MultivaluedMap;
+import jakarta.ws.rs.core.UriBuilder;
+import jakarta.ws.rs.ext.Provider;
+import org.openremote.model.Constants;
+import org.openremote.model.protocol.ProtocolUtil;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Supplier;
+import java.util.regex.Matcher;
+import java.util.stream.Collectors;
+
+import static org.openremote.container.web.QueryParameterInjectorFilter.DYNAMIC_TIME_PATTERN;
+import static org.openremote.container.web.QueryParameterInjectorFilter.DYNAMIC_VALUE_PROPERTY;
+
+@Provider
+@Priority(1)
```

Comment:
Because for some reason I am getting this error: `RESTEASY003720: path param time has not been provided by the parameter map` when the time filter is applied after the value filter. What I believe is causing this, is that the value filter actually adds the path parameters to the URI. When the value filter is ran before the time filter, then the path parameters that have dynamic time will be attempted to be inserted without injecting the dynamic time, causing that error (hence `time`, which is the text that we have to indicate dynamic time on the value).